src/Tickets/Custom_Tables/V1/WP_Query/WP_Meta_Query_Modifier.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Service for manipulating Meta Queries for Event Tickets + Custom Tables.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Custom_Tables\V1\WP_Query;
+ */
+
+namespace TEC\Tickets\Custom_Tables\V1\WP_Query;
+
+use TEC\Events\Custom_Tables\V1\Models\Occurrence;
+use TEC\Events_Pro\Custom_Tables\V1\Models\Provisional_Post;
+use Tribe__Tickets__Attendee_Repository;
+use Tribe__Tickets__Ticket_Repository;
+use WP_Query;
+
+/**
+ * Service for manipulating Meta Queries for Event Tickets + Custom Tables.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Custom_Tables\V1\WP_Query;
+ */
+class WP_Meta_Query_Modifier {
+
+	/**
+	 * A reference to the Attendee Repository.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Tickets__Attendee_Repository
+	 */
+	protected $attendee_repository;
+
+	/**
+	 * A reference to the Ticket Repository.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Tickets__Ticket_Repository
+	 */
+	protected $ticket_repository;
+
+	/**
+	 * A list of meta keys to watch out for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected $meta_keys = [];
+
+	/**
+	 * The Provisional Post object.
+	 *
+	 * @since TBD
+	 *
+	 * @var Provisional_Post
+	 */
+	protected $provisional_post;
+
+	/**
+	 * Post types we are watching for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected $post_types = [];
+
+	/**
+	 * Ticket Meta Query Modifier constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Attendee_Repository $attendee_repository
+	 * @param Tribe__Tickets__Ticket_Repository   $ticket_repository
+	 * @param Provisional_Post                    $provisional_post
+	 */
+	public function __construct(
+		Tribe__Tickets__Attendee_Repository $attendee_repository,
+		Tribe__Tickets__Ticket_Repository $ticket_repository,
+		Provisional_Post $provisional_post
+	) {
+		$this->attendee_repository = $attendee_repository;
+		$this->ticket_repository   = $ticket_repository;
+		$this->meta_keys           = array_merge(
+			array_values( $attendee_repository->attendee_to_event_keys() ),
+			array_values( $ticket_repository->ticket_to_event_keys() )
+		);
+		$this->post_types          = array_merge(
+			array_values( $ticket_repository->ticket_types() ),
+			array_values( $attendee_repository->attendee_types() )
+		);
+		$this->provisional_post    = $provisional_post;
+	}
+
+	/**
+	 * Detects and modifies WP_Query for ticket queries.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Query $query The WP_Query fetching posts.
+	 */
+	public function modify_tickets_meta_query( $query ) {
+		if ( ! $query instanceof WP_Query ) {
+			return;
+		}
+
+		$post_types = array_values( (array) $query->get( 'post_type' ) );
+
+		if ( ! count( array_intersect( $post_types, $this->post_types ) ) ) {
+			return;
+		}
+
+		foreach ( $this->meta_keys as $prefix ) {
```

Comment:
Instead of doing it here, I think it would make sense  to filter the `event_id` in the [repositories](https://github.com/the-events-calendar/event-tickets/blob/8d569126e399f71000273d752505cf506d07f807/src/Tribe/Ticket_Repository.php#L109-L111) (this is just an example) by filtering the `event_id` variable the moment it's added to the queries.

E.g.
```diff
        public function filter_by_event_not_in( $event_id ) {
+              $event_id = tec_event_normalize_event_id( $event_id );
		$this->by( 'meta_not_in', $this->ticket_to_event_keys(), $event_id );
	}
```

The `tec_event_normalize_event_id` would be a function that would deal with the fact that CT1 is active or not and, if active, would normalize the Event ID using the `Occurrence::normalize_id` function.

Something like this (the code might contain mistakes, I typed it directly here), in TEC:

```php
function tec_event_normalize_event_id( $event_id ) {
    if ( empty ( $event_id ) ) || ! is_numeric( $event_id ) ) { return $event_id; }

    $ct1_active = tribe()->getVar( 'ct1_fully_activated' );

    if( ! $ct1_active ) {
        return $event_id;
    }
    
    return TEC\Events\Custom_Tables\V1\Models\Occurrence::normalize_id( $event_id );
}
```

Reply:
Looking at the Ticket and Attendee repositories, I do not see a lot of places where similar logic would apply.

Reply:
I thought about going this route, opted not to do it because I wasn't sure where else the event ID's would be used (custom queries etc) which would still receive a provisional ID.

That said, I'll take your recommendation - there's no guarantee those queries would want the post ID instead of the provisional ID in the future.

Reply:
Went a slightly different way, this change does not require a TEC version dependency - sticks to a filter CT1 hooks on.